### ISSUE TITLE
testing

### ISSUE THREAD

**Issue by gravity1989** [mbhenaff/spectral-lib#6]
So I managed to train using the cifar dataset.

How do i test the model using the test dataset?

**Comment by mbhenaff** [comment]
Hi Rahul,

The test performance is computed line 69 of test/train.lua. The computePerf
function takes in a split ('train' or 'test') and computes performance of
the model on that split.

-Mikael

**Comment by gravity1989** [comment]
Thanks Mikael.

On Wed, Mar 30, 2016 at 7:10 AM, mbhenaff notifications@github.com wrote:

> Hi Rahul,
> 
> The test performance is computed line 69 of test/train.lua. The computePerf
> function takes in a split ('train' or 'test') and computes performance of
> the model on that split.
> 
> -Mikael
> 
> On Tue, Mar 29, 2016 at 1:52 AM, Rahul V notifications@github.com wrote:
> 
> > So I managed to train using the cifar dataset.
> > 
> > How do i test the model using the test dataest?
> > 
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly or view it on GitHub
> > https://github.com/mbhenaff/spectral-lib/issues/6
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/mbhenaff/spectral-lib/issues/6#issuecomment-203185327

## 

Regards,
R.V.

**Comment by wangg12** [comment]
@gravity1989 I have a question. Is the Laplacian matrix obtained from all the training data and fixed during training? If a single sample is a graph structured data(say represented as a weighted matrix), how should I do?

**Comment by gravity1989** [comment]
I am not sure if its fixed or otherwise. I have just about managed to runt
he code.

The .mat file is loaded and modifications could be happening which is
invisible to me.

On Fri, Apr 1, 2016 at 1:02 PM, Wang Gu notifications@github.com wrote:

> @gravity1989 https://github.com/gravity1989 I have a question. Is the
> Laplacian matrix obtained from all the training data and fixed during
> training? If a single sample is a graph structured data(say represented as
> a weighted matrix), how should I do?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/mbhenaff/spectral-lib/issues/6#issuecomment-204290660

## 

Regards,
R.V.

**Comment by mbhenaff** [comment]
Yes, the Laplacian matrix is fixed during training. It is first estimated
from the training data, and then its eigenvectors are used to define a
Graph Fourier Transform which is a generalization of the regular Fourier
transform. Pointwise multiplication in this new "frequency" space
corresponds to a convolution on the graph.
